src/crypto/crypto_cipher.h

```diff
@@ -249,7 +249,7 @@ class CipherJob final : public CryptoJob<CipherTraits> {
       v8::Local<v8::Value>* result) override {
     Environment* env = AsyncWrap::env();
     CryptoErrorStore* errors = CryptoJob<CipherTraits>::errors();
-    if (out_.size() > 0) {
+    if (out_.size() > 0 || errors->Empty()) {
```

Comment:
A few lines below, there's this:

```cpp
if (errors->Empty())
  errors->Capture();      // this is unreachable now
CHECK(!errors->Empty());
```

So, either those lines are not needed, or there is no guarantee that `errors->IsEmpty()` will return `false` only if no error occurred.

If `errors->IsEmpty()` is true if and only if the operation succeeded, then `out_.size() > 0` is not a required condition.

If `errors->IsEmpty()` is never `true` when the operation failed, then the lines below should be removed.

Reply:
Should I move `if (errors->Empty()) errors->Capture();` before `if (out_.size() > 0)`?

Reply:
I think @jasnell might be the best person to ask, but that might be enough to solve the problem. It would make the sanity check `CHECK(!errors->Empty())` below useless.

Reply:
I've moved it above.

/cc @jasnell